blocks/theano_expressions.py

```diff
@@ -14,11 +14,11 @@ def l2_norm(tensors):
         The tensors.
 
     """
-    flattened = [tensor.as_tensor_variable(t).flatten() for t in tensors]
-    flattened = [(t if t.ndim > 0 else t.dimshuffle('x'))
-                 for t in flattened]
-    joined = tensor.join(0, *flattened)
-    return tensor.sqrt(tensor.sqr(joined).sum())
+    grand_sum = 0.0
```

Comment:
We had quite a discussion once with @bartvm about the right way to sum here. The method you propose has a disadvantage that the computation graph will be as deep as many tensor are given as arguments. This might hurt at compile time and also makes it harder to investigate debugprints.

On the other hand, I agree that the old implementation is memory inefficient. 

It seems like one can combine the advantages of two by creating a vector of scalars in the way you do it and joining these scalars like we did. What do you think?